### PR TITLE
MAT-7013: Add cron job for midnight

### PR DESCRIPTION
**Pull request by mcmcphillips**
## Terminology Service PR

Jira Ticket: [MAT-7013](https://jira.cms.gov/browse/MAT-7013)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/test/java/gov/cms/madie/terminology/task/UpdateCodeSystemTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.*;
+
+import static org.mockito.Mockito.*;
+@ExtendWith(MockitoExtension.class)
```
Comment (jkotanchik-SB) [inline src/test/java/gov/cms/madie/terminology/task/UpdateCodeSystemTaskTest.java:16]: Minor editorial: empty line between imports and class body.

File: src/main/java/gov/cms/madie/terminology/task/UpdateCodeSystemTask.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  @Value("${code-system-refresh-task.terminology-key}")
+  private String apiKey;
+
+  @Scheduled(cron = "${code-system-refresh-task.code-system-cron-date-time}") // every midnight
+  public void updateCodeSystems() {
+    log.info("Starting scheduled task to update code systems.");
+
+    UmlsUser user = new UmlsUser();
+    user.setApiKey(apiKey);
+    ResponseEntity<List<CodeSystem>> response =
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/terminology/task/UpdateCodeSystemTask.java]: Since the ResponseEntity isn't returned, we can skip assigning it beyond the `List<CodeSystem>`.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -18,7 +18,10 @@ okta:
     issuer: ${OKTA_ISSUER:https://dev-18092578.okta.com/oauth2/default}
     audience: ${OKTA_AUDIENCE:api://default}
 
-admin-api-key: ${ADMIN_API_KEY:0a51991c}
+code-system-refresh-task:
+  admin-api-key: ${ADMIN_API_KEY:0a51991c}
+  terminology-key: ${VSAC_API_KEY:defaultvaluetopreventillegalappstate}
+  code-system-cron-date-time:  ${CS_REFRESH_TIME:@midnight}
```
Comment (chubert-sb) [inline src/main/resources/application.yml:24]: This is nice. I've never seen scheduler use that midnight tag before

**Review by chubert-sb — DISMISSED** [review]

**Review by ethankaplan — APPROVED** [review]
